### PR TITLE
Update README and other documentation cleanup

### PR DESCRIPTION
**Pull request by mr-c**
Include README in the Read The Docs site
Move the CLI options to a new page
Headings formatting fix
Fix udocker/singularity examples
Explicit section for Singularity

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -140,16 +145,25 @@ maintain which implementation ``cwl-runner`` points to via a symbolic file
...
-You may also want to have the following installed:
-- `node.js <https://nodejs.org/en/download/>`_
-- Docker, udocker, or Singularity (optional)
+  * Podman
+  * `Docker <https://docs.docker.com/engine/install/>`_
+  * Singularity/Apptainer: See `Using Singularity`_
+  * udocker: See `Using uDocker`_
+
+* `node.js <https://nodejs.org/en/download/>`_ for evaluating CWL Expressions quickly
+  (required for `udocker` users, optional but recommended for other container engines (podman/d
```
Comment (kinow) [inline README.rst:160]: I think the end, (podman/d is wrong? maybe other container engines e.g. podman) ?

**Comment by mr-c** [comment]
Odd, the table of contents looks much better when I render the docs locally

image

vs https://cwltool--1778.org.readthedocs.build/en/1778/

**Review by kinow — APPROVED** [review]
Just had a look at the table of contents, and it looks OK to me @mr-c 

https://cwltool--1778.org.readthedocs.build/en/1778/

:+1:

**Comment by mr-c** [comment]
Thanks @kinow !

**Comment by mr-c** [comment]
> Just had a look at the table of contents, and it looks OK to me @mr-c

Yeah, I had to pin the version of `sphinx-rtd-theme`; they default to an older version (presumably to not break older setups)

**Comment by mr-c** [comment]
I wanted to change some of the headings, but then that would break links.

In RST, I can add silent anchors, but GitHub strips them out https://github.com/github/markup/issues/1535
GitHub, however, does allow the use of the `::raw html` directive, but PyPI does not :-/

So I reverted all the header changes for now

**Pull request merged**